### PR TITLE
Updating documentation and examples

### LINKED ISSUE #932: [3.0.1-rc1] docs out of date
The docs and examples of the `cloud-init` disk are out of date.
Also check that the new `disks` is used in examples.

### PR DESCRIPTION
**Pull request by hestiahacker**
This closes #932 

- Updated examples to use the new disks syntax
- Added the `cloudini_cdrom_storage` (now a required argument) and boot order (to work around #901 until it is fixed)
- Updated documentation on the cloud_init_disk to have a `disks` block in the example and point to the easier way to use cloud-init
- Pointed to the example file in the cloud_init guide

### CODE REVIEW

File: docs/resources/cloud_init_disk.md [hunk 1]
```diff
@@ -47,12 +52,17 @@ EOT
...
-  disk {
-    type    = "scsi"
-    media   = "cdrom"
-    storage = local.iso_storage_pool
-    volume  = proxmox_cloud_init_disk.ci.id
-    size    = proxmox_cloud_init_disk.ci.size
+  disks {
+    scsi {
+      scsi0 {
+        disk {
```
Comment (Tinyblargon) [inline docs/resources/cloud_init_disk.md:58]: With the new disk schema it should be like this:

```hcl
cd {
  iso = "${local.iso_storage_pool}:${proxmox_cloud_init_disk.ci.id}
}
```

Not 100% sure what `local.iso_storage_pool` and `proxmox_cloud_init_disk.ci.id` are in this situation.

**Review by Tinyblargon — CHANGES_REQUESTED** [review]

**Comment by hestiahacker** [comment]
Thanks for the review and correction. I think it should be good to go now

**Comment by TAAGECH9** [comment]
@Tinyblargon , do you think this PR could be merged 😄? I was struggling on my own with exactly the same issue and wanted to create a PR and then realized that there is already this PR open.

**Comment by TAAGECH9** [comment]
@Tinyblargon ,that was quick 🔥. Thank you very much ❤️

**Pull request merged**